src/Components/Endpoints/src/Builder/ComponentTypeMetadata.cs

```diff
@@ -20,9 +20,25 @@ public ComponentTypeMetadata([DynamicallyAccessedMembers(Component)] Type compon
         Type = componentType;
     }
 
+    /// <summary>
+    /// Initializes a new instance of <see cref="ComponentTypeMetadata"/>.
+    /// </summary>
+    /// <param name="componentType">The component type.</param>
+    /// <param name="isStaticPage">A flag indicating whether the page's route is declared as static.</param>
+    public ComponentTypeMetadata([DynamicallyAccessedMembers(Component)] Type componentType, bool isStaticPage)
+        : this(componentType)
+    {
+        IsStaticRoute = isStaticPage;
+    }
+
     /// <summary>
     /// Gets the component type.
     /// </summary>
     [DynamicallyAccessedMembers(Component)]
     public Type Type { get; }
+
+    /// <summary>
+    /// Gets a flag indicating whether the page's route is declared as static.
+    /// </summary>
+    public bool IsStaticRoute { get; }
```

Comment:
This could go into an entirely separate metadata class but then that's an extra metadata entry we have to look up on every request, and I don't see there being much reason for it.